plugin/kubernetes/controller.go

```diff
@@ -151,10 +156,12 @@ func newdnsController(kubeClient kubernetes.Interface, opts dnsControlOpts) *dns
 
 	dns.nsLister, dns.nsController = cache.NewInformer(
 		&cache.ListWatch{
-			ListFunc:  namespaceListFunc(dns.client, dns.selector),
-			WatchFunc: namespaceWatchFunc(dns.client, dns.selector),
+			ListFunc:  namespaceListFunc(dns.client, dns.namespaceSelector),
```

Comment:
If I understand correctly this will break the existing behavior: if a global selector is set, coredns was filtering namespaces too based on the selecor. Now if someone does not set a namespace selector but set a global one, all namespaces will be included. I think if namespace selector is not set we should fallback to the global selector to preserve the former behavior.

Reply:
I think the existing behavior is broken. Because if the `labels` configuration is set for namespaces, then it requires that labels be set on both `namespaces` and `pods`/`endpoints`/`services`. We can certainly make it backwards compatible, but I just don't see how someone could effectively use `labels` to filter namespaces.

Reply:
Agreed, you would need to have the same label everywhere
But bugs tend to often be someone's feature (but yes, if they agree upstream, this behavior is a lot better)